fuel-vm/src/interpreter/memory.rs

```diff
@@ -334,11 +575,15 @@ pub(crate) fn try_update_stack_pointer(
     ssp: Reg<SSP>,
     hp: Reg<HP>,
     new_sp: Word,
+    memory: &mut Memory,
 ) -> SimpleResult<()> {
-    if new_sp >= *hp || new_sp < *ssp {
+    if new_sp < *ssp {
         Err(PanicReason::MemoryOverflow.into())
+    } else if new_sp > *hp {
```

Comment:
So before, we had a check `new_sp >= *hp`, while now it is  `new_sp >*hp`. Is it expected? Did I get right that stack is `[0..SP)` and the heap is `[HP..MEM_SIZE)` and it is okay that `SP == HP`?

Reply:
Exactly correct.